File: MovieApp/src/app/service/data/userprofile.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { UserProfile } from '../../model/UserProfile';
import { Observable } from 'rxjs';
import { API_URL } from '../../app.constants';

@Injectable({
  providedIn: 'root',
})
export class UserprofileService {
  constructor(private http: HttpClient) {}

  getUserProfile(username: string): Observable<UserProfile> {
    return this.http.get<UserProfile>(
      `${API_URL}/private/userProfile/getUserById/${username}`
    );
  }

  saveUserProfile(userProfile: UserProfile): Observable<UserProfile> {
    return this.http.post<UserProfile>(
      `${API_URL}/public/userProfile/addUser`,
      userProfile
    );
  }

  updateUserProfile(
    username: string,
    userProfile: UserProfile
  ): Observable<UserProfile> {
    return this.http.put<UserProfile>(
      `${API_URL}/private/userProfile/update/${username}`,
      userProfile
    );
  }
}

File: movie-backend/userprofile-service/src/main/java/com/user/userprofileservice/service/UserProfileService.java
package com.user.userprofileservice.service;




import com.user.userprofileservice.dto.UserProfileDto;
import com.user.userprofileservice.dto.UserProfileUpdateDto;
import com.user.userprofileservice.model.UserProfile;

public interface UserProfileService {

    UserProfile getUserProfileById(String username);

    UserProfile saveUserProfile(UserProfileDto userProfileDto);

    UserProfile updateUserProfile(UserProfileUpdateDto userProfileDto, String username);

}
